Lib/fontTools/misc/configTools.py

```diff
@@ -0,0 +1,186 @@
+"""
+Code of the config system; not related to fontTools or fonts in particular.
+
+The options that are specific to fontTools are in `fontTools.config`.
+"""
+import logging
+from dataclasses import dataclass
+from typing import Any, Callable, Dict, Iterator, MutableMapping, Union
+
+log = logging.getLogger(__name__)
+
+__all__ = [
+    "ConfigError",
+    "ConfigAlreadyRegisteredError",
+    "ConfigValueParsingError",
+    "ConfigValueValidationError",
+    "ConfigUnknownOptionError",
+    "Option",
+    "OPTIONS",
+    "register_option",
+    "Config",
+]
+
+
+class ConfigError(Exception):
+    """Base exception for the config module."""
+
+
+class ConfigAlreadyRegisteredError(ConfigError):
+    """Raised when a module tries to register a configuration option that
+    already exists.
+
+    Should not be raised too much really, only when developing new fontTools
+    modules.
+    """
+
+    def __init__(self, name):
+        super().__init__(f"Config option {name} is already registered.")
+
+
+class ConfigValueParsingError(ConfigError):
+    """Raised when a configuration value cannot be parsed."""
+
+    def __init__(self, name, value):
+        super().__init__(
+            f"Config option {name}: value cannot be parsed (given {repr(value)})"
+        )
+
+
+class ConfigValueValidationError(ConfigError):
+    """Raised when a configuration value cannot be parsed."""
+
+    def __init__(self, name, value):
+        super().__init__(
+            f"Config option {name}: value is invalid (given {repr(value)})"
+        )
+
+
+class ConfigUnknownOptionError(ConfigError):
+    """Raised when a configuration option is unknown."""
+
+    def __init__(self, name, value):
+        super().__init__(
+            f"Config option {name} is unknown (with given value {repr(value)})"
+        )
+
+
+@dataclass
+class Option:
+    name: str
```

Comment:
Cosimo: delete this attribute, not needed because it's already in the dictionary key of OPTIONS